### PR TITLE
docs: map ECS migration status and remaining work

### PR DESCRIPTION
**Pull request by DrJKL**
## Summary

Replace the in-progress ECS roadmap commentary with a concise implementation status and focused architecture audits.

## Changes

- **What**: Documents completed and remaining migration work, adds decision traceability and dedicated authority, lifecycle, mutation, identity, compatibility, verification, and documentation audits, and provides an executive summary.

## Review Focus

Validate that current implementation, transitional boundaries, and remaining work are separated accurately without changing ADR decisions.

### CODE REVIEW

File: docs/architecture/ecs/ecs-decision-traceability.md
Comment (christian-byrne) [inline docs/architecture/ecs/ecs-decision-traceability.md:1]: Can you check https://github.com/Comfy-Org/comfy-multi-player

**Review by christian-byrne — COMMENTED** [review]
This is helpful. Can we put them into something like docs/architecture/ecs perhaps? For files that track current development process, gaps remaining, audits, etc. maybe a new folder that contains more temporary docs is warranted.

File: docs/architecture/ecs/ecs-decision-traceability.md
Comment (DrJKL) [reply docs/architecture/ecs/ecs-decision-traceability.md:1]: Spoilers...

**Comment by DrJKL** [comment]
> This is helpful. Can we put them into something like docs/architecture/ecs perhaps? For files that track current development process, gaps remaining, audits, etc. maybe a new folder that contains more temporary docs is warranted.

docs/architecture/migration/*.md?

File: docs/architecture/ecs/ecs-lifecycle-audit.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+## Concrete sequences
+
+### Graph load and configure
+
+1. `LGraph.configure` dispatches `configuring`, starts the named-widget shadow
+   load, normalizes topology, and clears or detaches the old graph.
+2. It adopts the graph ID and clears root buckets or only the current owner.
+3. It registers links first, then reroutes, and normalizes subgraph definitions.
+4. It creates every node shell and calls `LGraph.add` before any node configure.
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply docs/architecture/ecs/ecs-lifecycle-audit.md:51]: No change needed for this finding. The lifecycle audit already documents the staged order explicitly: root links are registered, then reroutes, then subgraph definitions are normalized. That matches `LGraph.configure`.

The collision boundary is also covered at the relevant levels: `LGraph.test.ts` exercises duplicate node IDs across definitions through `configure`; `subgraphLinkIdentity.spec.ts` loads cross-definition link/reroute collisions in the browser; and `subgraphDeduplication.test.ts` covers sibling group and reroute collisions. Normalization uses the root registrations as reserved incumbents before any subgraph is created, so moving it ahead of root registration is unnecessary. Repeating the ordering or adding equivalent collision tests would duplicate the current audit and coverage.

File: docs/architecture/ecs/ecs-state-authority-audit.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+## Node shell
+
+- **Authority:** `useNodeDataStore`; one plain `NodeState` in a root-flat,
+  owner-indexed bucket.
+- **Compatibility view/mirror:** `LGraphNode` fields are accessors over
+  `_state`, the store-returned reactive proxy. This is one object, not a copy.
+- **Reads:** renderer components consume `NodeState`; legacy code reads
+  `LGraphNode.title`, `mode`, colors, flags, slot arrays, and related accessors.
+- **Writes:** class setters and direct proxy writes; `registerNodeState` adopts
+  the proxy. These are not serializable commands.
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply docs/architecture/ecs/ecs-state-authority-audit.md]: Fixed in `08ac408`. The audit now calls this one store-backed `NodeState` and names `NodeInputSlot` and `NodeOutputSlot` instances in `inputs` and `outputs` as the transitional exception.

File: docs/architecture/ecs/ecs-migration-summary.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+## Current result
+
+- `nodeDataStore` owns renderer-facing node shell state.
+- `linkStore` owns link topology and slot connectivity.
+- `rerouteStore` owns reroute chain state; membership is derived.
+- `widgetValueStore` owns widget values, render metadata, and order.
+- Yjs-backed `layoutStore` owns persistent node, group, and reroute geometry.
+- `badgeSystem` derives transient badge rows instead of storing them.
+- Graph lifecycle registers, transfers, and tears down migrated state across
+  root graphs and nested subgraph definitions.
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply docs/architecture/ecs/ecs-migration-summary.md]: Fixed in `08ac408`. The executive summary now limits teardown to clear paths and names retained widget entries after normal node removal. The migration-plan lifecycle row records both that exception and the weaker cleanup guarantees of `configure(keep_old)`.

File: docs/architecture/output-slot-connectivity.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ Date: 2026-07-06
 Status: Accepted (follow-up to the
 [link topology store](link-topology-store.md); the minimal, non-breaking
 slice of the deferred `SlotConnection` component work in the
-[ECS migration plan](ecs-migration-plan.md))
+[ECS migration plan](ecs/ecs-migration-plan.md))
```
[the remark this answers is not in the document]
Comment (DrJKL) [reply docs/architecture/output-slot-connectivity.md]: Fixed. The suggested closing-parenthesis removal was applied in `b70a627`; `08ac408` also changes the matching opening parenthesis after `Status: Accepted` to a semicolon so the status line remains grammatically balanced.

File: docs/architecture/ecs/ecs-decision-traceability.md
Comment (DrJKL) [reply docs/architecture/ecs/ecs-decision-traceability.md:1]: Checked. `Comfy-Org/comfy-multi-player` implements the shared Yjs workflow document, stamped op applier, and canonical projection for the server-host/browser-follower model. I updated the traceability audit to distinguish that cross-repository contract from this frontend branch: the frontend does not yet depend on the package, and layout remains a separate frontend document by that package’s own contract.

**Comment by DrJKL** [comment]
Addressed the organization feedback by moving the ten migration-status and audit references under `docs/architecture/ecs/`. Durable target documents such as `ecs-target-architecture.md` and `ecs-lifecycle-scenarios.md` remain at the architecture root; inbound links were updated.

**Review by benjcooley — APPROVED** [review]
Approving. Docs-only change targeting `feature/ecs-migration` — no code risk. The consolidation into `docs/architecture/ecs/` addresses the folder-structure feedback, and the status write-up honestly separates what's shipped (nodeDataStore, linkStore, rerouteStore, widgetValueStore, Yjs layoutStore, badgeSystem) from what remains (command coverage beyond layout, cross-store transaction boundaries, snapshot-based undo/redo, class-based slots/widgets). CodeRabbit's actionable findings look addressed in b70a627/08ac408.
